docs/DeveloperGuide.adoc

```diff
@@ -358,12 +358,131 @@ image::UndoRedoActivityDiagram.png[width="650"]
+A user is not able to assign a remark to a person when adding the person. Therefore, all people in the CRM is originally without any remark by default.
+
+The implementation of this command is much like the `EditCommand`.
+For example, when a user type `remark 2 r\Likes to swim`. Then the remark field of the 2nd person in the current list will be changed to `Likes to swim`. It’s similar to editing a person’s phone number or address.
+
+==== Design Considerations
+
+===== Aspect: Implementation of `RemarkCommand`
+
+* **Alternative 1 (current choice):** Add a new `remark` command
+** Pros: Treat it as a command may be easy to understand. Only a few people needs a remark.
+** Cons: Cannot add a person with remark.
+* **Alternative 2:** Just make it a part of `EditCommand` and `AddCommand`
+** Pros: Only need to add remark field to every related class. We can add a person with remark.
+** Cons: We may have to type too many things when adding a person. Also, some people don’t have any remark.
+// end::remark[]
+
+// tag::colortag[]
+=== Color Tag feature
+==== Current Implementation
+
+To allow easy identification of different tags, a color tag feature is introduced.
+
+A string array is declared in PersonCard.java to include all the different colors for tags. These colors are defined, with their backgroud color and text color spelled out, in the the DarkTheme.css file.
+
+A hash code of the tag name is used to select a color for the tag such that it would remain consistent between different runs of the software.
+
+==== Design Considerations
+
+===== Aspect: Implementation of color tags
+
+* **Alternative 1 (current choice):** Assign a color from a predefined list based on tag name
+** Pros: No additional commands are needed to generate a color for the tags.
+** Cons: User cannot choose a color for the tag.
+* **Alternative 2:** Allow user to set a color for the tag
+** Pros: Allows more freedom for user customization.
+** Cons: Takes more time to implement color tags.
+// end::remark[]
+
+// tag::leadcontact[]
+=== Leads and Contacts feature
+==== Current Implementation
+
+People in the address book were stored under a single class with no options for differentiating between two types of classes.
+
+To implement a CRM, Leads and Contacts must be differentiated so that they can each hold different fields like a proper CRM. To do so, they currently extend the Person class so that all existing functionality associated with Persons will work with both Leads an Contacts. Unique fields will be given their own classes like the original generic fields like Name and Address.
+
+In a CRM, users start by adding Leads, who are people they have not sold to yet. As they begin selling, the people who they have sold to become Contacts.
+
+Many existing commands like AddCommand have been modified to fit the new paradigm so that it only adds Leads.
+
+A new conversion command ConvertCommand has been added to convert Leads into Contacts
```

Comment:
Add the relevant code and diagrams of how this is achieved. Remember to also update the Model diagram to account for this new Lead and Contact classes in the Person class.